### ISSUE TITLE
Example usage is confusing

### ISSUE THREAD

**Issue by Sycor4x** [NegatioN/OnlineMiningTripletLoss#3]
From the README, the example usage is given as

```
from triplet_loss import batch_hard_triplet_loss

labels = torch.randint(5) # our five labels

embeddings = model(labels)

loss = batch_hard_triplet_loss(labels, embeddings, margin=0.2)
loss.backward()
# and so on
```

Is this repository oriented around models which produce embeddings of labels? Or is the line `model(labels)` purely evocative?

It seems that a more standard usage would be to provide some kind of data (such as an image) and receive embeddings; then the triplet loss operates on the embeddings and labels together to determine the loss.

**Comment by NegatioN** [comment]
What I'm trying to say here, is that you need to have an embedding, and some sort of label for that embedding. I don't know how you would do it exactly with faces, but you're probably trying to which person that is, correct? The readme just contains the simplest possible example. :)

**Comment by Sycor4x** [comment]
Yes, using an embedding on faces is the canonical example. So in that setting, I'd provide the `faces` (not `labels`) to the model, and then use `labels` and `embeddings` as inputs to `batch_hard_triplet_loss`. It seems like 
```

from triplet_loss import batch_hard_triplet_loss

labels = torch.randint(5) # our five labels
faces = torch.randn(5,28,28)
embeddings = model(faces)

loss = batch_hard_triplet_loss(labels, embeddings, margin=0.2)
loss.backward()
# and so on
```

would make this usage more clear, if we explain that `faces` is a tensor of 5 greyscale images, each 28 by 28.

**Comment by NegatioN** [comment]
I would gladly accept a PR for this 😄
